### PR TITLE
Remove jobs.Srv and other jobs-related globals

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
Removes jobs.Srv and other jobs-related globals.

Enterprise PR: https://github.com/mattermost/enterprise/pull/207

#### Ticket Link
N/A

#### Checklist
N/A

### CODE REVIEW

**Comment by grundleborg** [comment]
As with the enterprise changes, this needs a rebase too.

**Comment by ccbrown** [comment]
@grundleborg rebased.

File: app/app.go [hunk 1]
```diff
@@ -25,6 +27,8 @@ type App struct {
 	Hubs                        []*Hub
 	HubsStopCheckingForDeadlock chan bool
 
+	Jobs *jobs.JobServer
```
Comment (coreyhulen) [inline app/app.go]: minor nit, why is this plural?  isn't it a single job server?

**Comment by ccbrown** [comment]
Ugh. It'll be easier to fix the build after my other PR is merged and another rebase.

File: app/app.go (hunk 1 above)
Comment (ccbrown) [reply app/app.go]: Single server, multiple jobs. It just seemed more natural to go from `jobs.DoThing` to `a.Jobs.DoThing`.

**Pull request merged**